File: server/Audi/Entities/Order.cs
using System;
using System.Collections.Generic;
using Audi.Models;

namespace Audi.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public string OrderNumber { get; set; }
        public string Email { get; set; }
        public AppUser User { get; set; }
        public Address BillingAddress { get; set; }
        public Address ShippingAddress { get; set; }
        public decimal TotalPrice { get; set; }
        public string CreditCardLast4Digit { get; set; }
        public string CreditCardType { get; set; } // Visa, MasterCard, AmericanExpress, etc
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
        public ICollection<OrderItem> OrderItems { get; set; }
        public string TrackingNumber { get; set; }
        public OrderStatus[] PreviousStatuses { get; set; }
        public string CurrentStatus { get; set; } // placed | shipped | delived | canceled - see OrderStatus.css
        public string CustomerNotes { get; set; }
        public string InternalNotes { get; set; }
        public string ShippingMethod { get; set; } // standard | expedited
        public decimal ShippingFee { get; set; } // 0 | 200
    }
}

File: client/projects/data/src/lib/models/order.ts
import { OrderStatus } from '../enums';
import { Address } from './address';
import { OrderItem } from './order-item';

export interface Order {
  id: number;
  orderNumber: string;
  userId: number | null;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  billingAddress: Address;
  shippingAddress: Address;
  totalPrice: number;
  creditCardLast4Digit: string;
  creditCardType: string;
  createdAt: Date;
  lastUpdated: Date;
  orderItems: OrderItem[];
  trackingNumber: string;
  previousStatuses: OrderStatusHistory[];
  currentStatus: OrderStatus;
  customerNotes: string;
  internalNotes: string;
}

export interface OrderStatusHistory {
  status: OrderStatus;
  updatedAt: Date;
}
